src/packages/hex.gleam

```diff
@@ -3,41 +3,204 @@ import gleam/json
 import gleam/http
 import gleam/result
 import gleam/hackney
+import gleam/list
 import gleam/dynamic.{Decoder} as d
+import time/parse.{parse_iso8601_to_gregorian_seconds}
+import gleam/pgo
+import gleam/string
+import gleam/int
+import gleam/option.{Option}
 
 pub type Error {
   HackneyError(hackney.Error)
   JsonError(json.DecodeError)
+  DatabaseError(pgo.QueryError)
+  ItemNotInListError(Nil)
 }
 
-pub fn query() {
+pub fn query(db: pgo.Connection) {
+  try last_scanned = get_last_scanned(db)
+
+  try packages = query_all_packages([], 1, last_scanned)
+
+  // Only update last scanned at if packages were actually
+  // found that way we only ever use the time from
+  // Hex's API 
+  case packages
+  |> list.length >= 1 {
+    True -> {
+      let len =
+        packages
+        |> list.length
+      assert Ok(last_package) =
+        packages
+        |> list.at(len - 1)
+        |> result.map_error(ItemNotInListError)
+      assert Ok(_) =
+        update_last_scanned(db, parse.parse_iso8601(last_package.updated_at))
+      Ok(Nil)
+    }
+    _ -> Ok(Nil)
+  }
+
+  packages
+  |> list.filter_map(sort_packages)
+  |> io.debug
```

Comment:
Debug statement left in here

Reply:
Will remove that at the end once this function is used else where, as for now need too see all the packages